Adyen/UI/Form/Items/SelectableFormItem/SelectableFormItemView.swift

```diff
@@ -0,0 +1,164 @@
+        stackView.setCustomSpacing(16, after: imageView)
+        stackView.spacing = 8
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.preservesSuperviewLayoutMargins = true
+        stackView.isLayoutMarginsRelativeArrangement = true
+        stackView.setContentHuggingPriority(.required, for: .vertical)
+        stackView.axis = .horizontal
+        stackView.alignment = .center
+        stackView.distribution = .fill
+
+        return stackView
+    }()
+
+    // MARK: - Item Button
+
+    private lazy var itemButton: UIButton = {
+        let customButton = UIButton(type: .custom)
+        customButton.addTarget(self, action: #selector(didSelectItemButton), for: .touchUpInside)
+        customButton.preservesSuperviewLayoutMargins = true
+        customButton.translatesAutoresizingMaskIntoConstraints = false
+        customButton.accessibilityIdentifier = item.identifier.map {
+            ViewIdentifierBuilder.build(scopeInstance: $0, postfix: "button")
+        }
+
+        customButton.addSubview(contentStackView)
+        contentStackView.isUserInteractionEnabled = false
+        contentStackView.adyen.anchor(inside: customButton)
+
+        return customButton
+    }()
+
+    @objc internal func didSelectItemButton() {
+        item.selectionHandler?()
+    }
+
+    /// Initializes the selectable form item view.
+    public required init(item: SelectableFormItem) {
+        super.init(item: item)
+        backgroundColor = item.style.backgroundColor
+
+        addSubview(itemButton)
+
+        accessibilityIdentifier = item.identifier
+        accessibilityLabel = item.accessibilityLabel
+        isAccessibilityElement = true
+
+        updateIcon()
+        updateImageView(style: item.style)
+        configureConstraints()
+
+        preservesSuperviewLayoutMargins = true
+
+        itemButton.adyen.anchor(inside: self.layoutMarginsGuide)
+
+        observe(item.$isSelected) { [weak self] isSelected in
+            guard let self else { return }
+            self.checkmarkImageView.isHidden = !isSelected
+        }
+    }
+
+    override public func didMoveToWindow() {
+        super.didMoveToWindow()
+        updateIcon()
+    }
+
+    override public func layoutSubviews() {
+        super.layoutSubviews()
+        imageView.adyen.round(using: item.style.imageStyle.cornerRounding)
+    }
+
+    private func updateImageView(style: SelectableFormItemStyle) {
+        imageView.contentMode = style.imageStyle.contentMode
+        imageView.clipsToBounds = style.imageStyle.clipsToBounds
+        imageView.layer.borderWidth = style.imageStyle.borderWidth
+        imageView.layer.borderColor = style.imageStyle.borderColor?.cgColor
+    }
+
+    private func updateIcon() {
+        if let iconUrl = item.imageUrl, window != nil {
+            imageLoadingTask = imageView.load(url: iconUrl, using: imageLoader)
+        } else {
+            imageLoadingTask = nil
```

Comment:
Should we cancel loading first?

Reply:
edd71e566cf4d9d230b1c2b66769351442f6f61b